x/service/keeper/update_relay_mining_difficulty.go

```diff
@@ -53,12 +44,19 @@ func (k Keeper) UpdateRelayMiningDifficulty(
 	// Iterate over the relaysPerServiceMap deterministically by sorting the keys.
 	// This ensures that the order of the keys is consistent across different nodes.
 	// See comment: https://github.com/pokt-network/poktroll/pull/840#discussion_r1796663285
+	targetNumRelays := k.GetParams(ctx).TargetNumRelays
 	sortedRelayPerServiceMapKeys := getSortedMapKeys(relaysPerServiceMap)
 	for _, serviceId := range sortedRelayPerServiceMapKeys {
 		numRelays := relaysPerServiceMap[serviceId]
 		prevDifficulty, found := k.GetRelayMiningDifficulty(ctx, serviceId)
 		if !found {
-			prevDifficulty = NewDefaultRelayMiningDifficulty(ctx, logger, serviceId, numRelays)
+			prevDifficulty = NewDefaultRelayMiningDifficulty(
+				ctx,
+				logger,
+				serviceId,
+				numRelays,
```

Comment:
https://discord.com/channels/824324475256438814/1138895490331705354/1307077093351362621

```suggestion
				targetNumRelays,
```

(cc @red-0ne )

Reply:
This change causes some test failures:

```
=== RUN   TestComputeNewDifficultyHash_MonotonicallyIncreasingRelays
    update_relay_mining_difficulty_test.go:45: Relay Mining Increasing Difficult Debug. NumRelays: 1000, EMA: 90099, TargetHash: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
    update_relay_mining_difficulty_test.go:34: 
        	Error Trace:	/home/bwhite/Projects/pokt/poktroll/x/service/keeper/update_relay_mining_difficulty_test.go:34
        	Error:      	"10000" is not greater than "90099"
        	Test:       	TestComputeNewDifficultyHash_MonotonicallyIncreasingRelays
--- FAIL: TestComputeNewDifficultyHash_MonotonicallyIncreasingRelays (0.00s)
```
```
=== RUN   TestUpdateRelayMiningDifficulty_Base
    update_relay_mining_difficulty_test.go:105: 
        	Error Trace:	/home/bwhite/Projects/pokt/poktroll/x/service/keeper/update_relay_mining_difficulty_test.go:105
        	Error:      	Not equal: 
        	            	expected: 0x3e8
        	            	actual  : 0x15ff3
        	Test:       	TestUpdateRelayMiningDifficulty_Base
--- FAIL: TestUpdateRelayMiningDifficulty_Base (0.00s)

Expected :0x3e8
Actual   :0x15ff3
```
```
=== RUN   TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_below_target
    update_relay_mining_difficulty_test.go:217: 
        	Error Trace:	/home/bwhite/Projects/pokt/poktroll/x/service/keeper/update_relay_mining_difficulty_test.go:217
        	Error:      	Not equal: 
        	            	expected: 0x64
        	            	actual  : 0x15f99
        	Test:       	TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_below_target
--- FAIL: TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_below_target (0.00s)

Expected :0x64
Actual   :0x15f99
```
```
=== RUN   TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_above_target
    update_relay_mining_difficulty_test.go:217: 
        	Error Trace:	/home/bwhite/Projects/pokt/poktroll/x/service/keeper/update_relay_mining_difficulty_test.go:217
        	Error:      	Not equal: 
        	            	expected: 0x5f5e100
        	            	actual  : 0x99f610
        	Test:       	TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_above_target
--- FAIL: TestUpdateRelayMiningDifficulty_FirstDifficulty/First_Difficulty_way_above_target (0.00s)

Expected :0x5f5e100
Actual   :0x99f610
```
```

Reply:
All tests are passing for me.